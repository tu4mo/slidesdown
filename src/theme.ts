export default {
  breakpoints: {
    md: '576px',
    xl: '768px',
  },
} as const
