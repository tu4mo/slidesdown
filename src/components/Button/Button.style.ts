import styled from 'styled-components'
import theme from '../../theme'

export const StyledButton = styled.button`
  background-color: #fff;
  border: 1px solid var(--color-purple);
  color: var(--color-purple);
  cursor: pointer;
  font-family: var(--font-default);
  font-size: 0.75rem;
  font-weight: bold;
  line-height: 2.375rem;
  outline: none;
  padding: 0 0.75rem;
  white-space: nowrap;

  @media (min-width: ${theme.breakpoints.md}) {
    font-size: 1rem;
    padding: 0 1rem;
  }

  &:focus {
    outline: 1px solid var(--color-purple);
    outline-offset: 1px;
  }

  &:disabled {
    border-color: var(--color-border);
    color: var(--color-border);
    cursor: default;
  }

  &:hover:not(:disabled) {
    background-color: var(--color-purple);
    color: #fff;
  }
`
