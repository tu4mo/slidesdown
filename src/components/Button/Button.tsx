import { ReactNode } from 'react'

import { StyledButton } from './Button.style'

interface ButtonProps {
  children: ReactNode
  disabled?: boolean
  onClick(): void
}

const Button = ({ children, disabled, onClick }: ButtonProps) => (
  <StyledButton
    disabled={disabled}
    onClick={onClick}
  >
    {children}
  </StyledButton>
)

export default Button
