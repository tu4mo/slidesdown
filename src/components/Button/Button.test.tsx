import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { ThemeProvider } from 'styled-components'

import theme from '../../theme'
import Button from './Button'

it('renders correctly', () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={theme}>
        <Button onClick={() => {}}>Test</Button>
      </ThemeProvider>
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})
