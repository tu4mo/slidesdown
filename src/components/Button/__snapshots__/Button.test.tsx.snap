// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  background-color: #fff;
  border: 1px solid var(--color-purple);
  color: var(--color-purple);
  cursor: pointer;
  font-family: var(--font-default);
  font-size: 0.75rem;
  font-weight: bold;
  line-height: 2.375rem;
  outline: none;
  padding: 0 0.75rem;
  white-space: nowrap;
}

.c0:focus {
  outline: 1px solid var(--color-purple);
  outline-offset: 1px;
}

.c0:disabled {
  border-color: var(--color-border);
  color: var(--color-border);
  cursor: default;
}

.c0:hover:not(:disabled) {
  background-color: var(--color-purple);
  color: #fff;
}

@media (min-width:576px) {
  .c0 {
    font-size: 1rem;
    padding: 0 1rem;
  }
}

<button
  className="c0"
  onClick={[Function]}
>
  Test
</button>
`;
